_drafts/arm-cortex-m-exceptions-and-nvic.md

```diff
@@ -0,0 +1,699 @@
+---
+title: "Cortex-M Exception Handling"
+description: "Step-by-step walkthrough of ARM Cortex-M Exception handlers & how to configure ARM
+System Fault Handlers and the Nested Vector Interrupt Controller (NVIC) with examples in C"
+author: chris
+---
+
+Nearly all embedded systems at one point or another rely on the ability to handle asynchronous
+events. Whether it be external sensor data from an accelerometer for tracking steps or periodic
+timer events to trigger a context switch for an RTOS
+
+<!-- excerpt start -->
+
+In this article we will dive into the details of how the ARM Cortex-M _exception model_ supports
+the handling of asynchronous events. We
+will walk through different exception types supported, terminology (i.e NVIC, ISR, Priority), the
+configuration registers used, common configuration settings and a few examples written in C.
+
+<!-- excerpt end -->
+
+Note: For the most part, the exception handling mechanism in all Cortex-M processors (ARMv6-M, ARMv7-M, ARMv8-M architectures) is the same. I will point out differences that do arise in the relevant sections below.
+
+## Table of Contents
+
+<!-- prettier-ignore -->
+* auto-gen TOC:
+{:toc}
```

Comment:
The TOC is a bit of a mess

Reply:
I'm going to update to nest the registers in one section and I'm dropping the "Advanced topics" section altogether. Let me know if you think other parts look messy!